File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Statements/WhileStatement.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

using CodeHelper.Core.Parse.ParseResults;
using CodeHelper.Core.Parse.ParseResults.Workflows.Expressions;
using CodeHelper.Core.Parse.ParseResults.Workflows;
namespace CodeHelper.Core.Parse.ParseResults.Workflows.Statements
{

    /**
     *
     * @author Administrator
     */
    public class WhileStatement : TokenPair
    {
        public Expression Condition { get; set; }
        public State State { get; set; } 

        public WhileStatementInfo Parse()
        {
            var rslt = new WhileStatementInfo(this);
            rslt.Condition = this.Condition.Parse();
            rslt.Statements = State.Parse(null);
            return rslt;
        }
    }
}

File: ToolBag/ParseTool/src/parsetool/workflow/models/statements/WhileStatement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package parsetool.workflow.models.statements;

import parsetool.models.common.TokenPair;
import parsetool.workflow.models.State;
import parsetool.workflow.models.expressions.Expression;

/**
 *
 * @author Administrator
 */
public class WhileStatement extends TokenPair {
    private Expression condition;
    private State state;

    /**
     * @return the condition
     */
    public Expression getCondition() {
        return condition;
    }

    /**
     * @param condition the condition to set
     */
    public void setCondition(Expression condition) {
        this.condition = condition;
    }

    /**
     * @return the state
     */
    public State getState() {
        return state;
    }

    /**
     * @param state the state to set
     */
    public void setState(State state) {
        this.state = state;
    }
}
